File: flutter/packages/workern_models/lib/billing/billing_interval_unit.dart
enum BillingIntervalUnit { day, week, month, year }

extension BillingIntervalUnitX on BillingIntervalUnit {
  String get value {
    switch (this) {
      case BillingIntervalUnit.day:
        return 'DAY';
      case BillingIntervalUnit.week:
        return 'WEEK';
      case BillingIntervalUnit.month:
        return 'MONTH';
      case BillingIntervalUnit.year:
        return 'YEAR';
    }
  }

  static BillingIntervalUnit fromValue(String value) {
    return BillingIntervalUnit.values.firstWhere(
      (unit) => unit.value == value,
      orElse: () => BillingIntervalUnit.month,
    );
  }
}

File: libs/shared/models/src/lib/interfaces/billing/billing-interval-unit.ts
export enum BillingIntervalUnit {
  DAY = 'DAY',
  WEEK = 'WEEK',
  MONTH = 'MONTH',
  YEAR = 'YEAR'
}

export function getBillingIntervalUnitInMs(
  intervalUnit: BillingIntervalUnit
): number {
  switch (intervalUnit) {
    case BillingIntervalUnit.DAY:
      return 24 * 60 * 60 * 1000; // 1 day in milliseconds
    case BillingIntervalUnit.WEEK:
      return 7 * 24 * 60 * 60 * 1000; // 1 week in milliseconds
    case BillingIntervalUnit.MONTH:
      return 30 * 24 * 60 * 60 * 1000; // Approx. 1 month in milliseconds
    case BillingIntervalUnit.YEAR:
      return 365 * 24 * 60 * 60 * 1000; // Approx. 1 year in milliseconds
    default:
      throw new Error(`Unsupported billing interval unit: ${intervalUnit}`);
  }
}
